GameServerLib/GameObjects/AttackableUnits/AttackableUnit.cs

```diff
@@ -1398,6 +1400,40 @@ public void SetStatus(StatusFlags status, bool enabled)
             }
         }
 
+        public void UpdateStatus()
+        {
+            // Combine the status effects of all the buffs
+            Dictionary<StatusFlags, bool> finalEffects = new Dictionary<StatusFlags, bool>();
+            foreach (IBuff buff in GetBuffs())
```

Comment:
This is in general not very efficient. We should have min priority queue based on the buff duration and its status effects. I'll open an issue and we can implement it later.